runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/MemoryAllocationException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.nemo.runtime.executor.data;
+
+/**
+ * An exception to be thrown when a memory allocation operation is not successful.
+ * Cases where allocation is not successful are when the allocated memory exceeds the
+ * amount specified by the job configuration or
+ */
+public class MemoryAllocationException extends Exception {
+
+  public MemoryAllocationException() {
+    super();
+  }
+
+  public MemoryAllocationException(final String message, final Throwable cause) {
+    super(message, cause);
+  }
+
+  public MemoryAllocationException(final String message) {
```

Comment:
Other methods except this one are unused.
Perhaps the `message` argument can be removed, and something like `super("Trying to allocate memory that exceeds JobConf. MaxOffheapMb")` can be used.

Reply:
I will remove the unused methods :) And for the message, our usages include different messages like "pool empty", "lack of memory" or "exceeds MaxOffHeapMb". So maybe we should keep it?